internal/argocd/health.go

```diff
@@ -170,7 +170,7 @@ func (h *applicationHealth) GetApplicationHealth(
 	// considered unhealthy as they may indicate a problem which can result in
 	// e.g. the health status result to become unreliable.
 	if errConditions := filterAppConditions(app, healthErrorConditions...); len(errConditions) > 0 {
-		issues := make([]error, len(errConditions))
+		issues := make([]error, 0, len(errConditions))
```

Comment:
Nice catch. But it's not _this_ line that's wrong. It's what comes after it. This is what it should have been:

```golang
		issues := make([]error, len(errConditions))
		for i, condition := range errConditions {
			issues[i] = fmt.Errorf(
				"Argo CD Application %q in namespace %q has %q condition: %s",
				appKey.Name,
				appKey.Namespace,
				condition.Type,
				condition.Message,
			)
		}
```

Reply:
Similar for the other spot.

Reply:
Modified.

Please review again.